src/snowflake/cli/api/project/schemas/template.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import Any, List, Optional
+
+from pydantic import BaseModel, Field
+from snowflake.cli.api.secure_path import SecurePath
+
+
+class TemplateVariableType(Enum):
+    STRING = "string"
+    INTEGER = "int"
+    FLOAT = "float"
+
+    @property
+    def python_type(self):
+        return {
+            TemplateVariableType.STRING: str,
+            TemplateVariableType.INTEGER: int,
+            TemplateVariableType.FLOAT: float,
+        }[self]
+
+
+class TemplateVariable(BaseModel):
+    name: str = Field(..., title="Variable identifier")
+    type: Optional[TemplateVariableType] = Field(  # noqa: A003
```

Comment:
Do we need enum at all? `TemplateVariableType = Union[str, float, int]`? Is there any value added in the enum?

Reply:
I've just checked - If we do that, TemplateVariable.type would expect basically any value of typ str, int or float. We wanted choice field between str/int/float/None
Enum does validation for us

Reply:
Changed to `Literal["string", "int", "float"]`, validation works well